### PR TITLE
`config auto` now generates a config file even when it encounters an error

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #3284 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-types/src/core/ics24_host/identifier.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ use crate::core::ics24_host::error::ValidationError;
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 #[serde(from = "tendermint::chain::Id", into = "tendermint::chain::Id")]
 pub struct ChainId {
-    id: String,
+    pub id: String,
```
Comment (romac) [inline crates/relayer-types/src/core/ics24_host/identifier.rs:21]: You can call `as_str()` in `ChainId` instead of accessing the `id` field directly

Comment (seanchen1991) [reply crates/relayer-types/src/core/ics24_host/identifier.rs]: I specifically wanted the name of the chain, i.e. everything before the dash. It looks like the `id` field already includes both the chain name and the version, so I added a `name` method in [9f63875](https://github.com/informalsystems/hermes/pull/3466/commits/9f63875e205e2bddf241cb8c97c90955163bc733)

File: crates/relayer-cli/src/commands/config/auto.rs [hunk 2]
```diff
@@ -75,59 +76,102 @@ impl Runnable for AutoCmd {
...
                         "Config file written successfully at '{}'",
                         self.path.display()
                     ))
-                    .exit(),
-                    Err(e) => Output::error(e).exit(),
+                    .exit()
+                } else {
+                    Output::success_msg(format!(
+                        "Config file written successfully at '{}'.
+                        However, configurations for the following chains were not able to be generated: {:?}",
```
Comment (romac) [inline crates/relayer-cli/src/commands/config/auto.rs:163]: Nice!

**Review by romac — APPROVED** [review]
Great work!

**Pull request merged**